students/srmorehouse/Lesson06/mailroom4.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+
+"""
+Steve Morehouse
+Lesson 06
+"""
+
+donor_db = {"William Gates, III": [653772.32, 12.17],
+            "Jeff Bezos": [877.33],
+            "Paul Allen": [663.23, 43.87, 1.32],
+            "Mark Zuckerberg": [1663.23, 4300.87, 10432.0]
+            }
+
+prompt = "\n".join(("Welcome to the Mailroom!",
+          "Please choose from below options:",
+          "1 - Send a Thank You",
+          "2 - Create Report",
+          "3 - Send letters to all",
+          "4 - Exit",
+          ">>> "))
+
+
+"""
+list all donors
+"""
+def list_donors ():
+    list = []
+    for i in donor_db.values():
+        list.append(i[0])
+    [print(donor) for donor in donor_db]
+
+"""
+If the user (you) selects “Send a Thank You” option, prompt for a Full Name.
+If the user types list show them a list of the donor names and re-prompt.
+If the user types a name not in the list, add that name to the data structure and use it.
+If the user types a name in the list, use it.
+Once a name has been selected, prompt for a donation amount.
+Convert the amount into a number; it is OK at this point for the program to crash if
+someone types a bogus amount.
+Add that amount to the donation history of the selected user.
+Finally, use string formatting to compose an email thanking the donor for their generous donation.
+Print the email to the terminal and return to the original prompt.
+It is fine (for now) for the program not to store the names of the new donors that had been added,
+in other words, to forget new donors once the script quits running.
+"""
+
+def send_a_thank_you ():
+    while True:
+        name = input("Type 'list' to see a list of names or enter a name: ")
+        # now redirect to feature functions based on the user selection
+        if name == "list":
+            list_donors()
+        elif not any(name in i for i in donor_db.values()):
+            donor_db.update({name: []})
+            break
+        else:
+            for i in range (len (donor_db)):
+                if donor_db[i][0] == name:
+                    break
+
+    # enter donation amount
+    donation = float(input("Enter a donation amount: "))
+    donor_db[name].append(donation)
+
+    # write email
+    msg = f"\n{name},\n\nThank you for your donation of ${donation}.\n"
+    print(msg)
```

Comment:
This is having the same issue as your mailroom2 file, isn't allowing me to add new donations for donors - it's overwriting each donation.